molecule/common/verify.yml

```diff
@@ -0,0 +1,52 @@
+---
+- name: Start the QDR
+  hosts: qdrtest
+  tasks:
+    - name: Check for qdrouterd.conf
+      find:
+        paths: /etc/qpid-dispatch/
+        patterns: qdrouterd.conf
+      register: conf
+      failed_when: conf.files|length != 1
+
+    - name: Start qdrouterd
+      command: qdrouterd -U qdrouterd -d
+
+- name: Functional tests
+  hosts: localhost
+  tasks:
+    - block:
+      - name: Clear the failure flag
+        set_fact: functest_failed_flag="unknown"
+
+      - name: Run the qdr test from STF functional-tests
+        include_role:
+          name: functional_tests
+          tasks_from: test_qdr
+        vars:
+          qdr_container_name: 'qdrtest'
+
+      rescue:
+        - name: Set flag if functional tests failed
+          set_fact: functest_failed_flag="failed"
+
+- name: Process qdr log
+  hosts: qdrtest
+  tasks:
+    - name: Slurp the file
+      slurp:
+        src: /var/log/qdrouterd/qdrouterd.log
+      register: qdrlog
+
+    - name: Decode the content
+      set_fact:
+        qdrlog_content: "{{ qdrlog['content'] | b64decode }}"
+
+    - name: Split into lines
+      debug:
+        msg: "{{ qdrlog_content.split('\n') }}"
+
+    - name: Check for errors
+      debug:
+        msg: "[WARNING] Found errors in the log. Please review above"
```

Comment:
Do any reviewers know how to whip up the custom callback module we need that will allow me to print a message like this in Orange/Red instead of green? I don't think an error in the log should fail a test that is otherwise passing, but I do want to draw attention to them visually.

Reply:
What about just doing `ignore_errors: true` ? Intent is just to check for errors and continue on even on "failure" right?

Reply:
I found this: https://medium.com/opsops/how-to-print-warning-from-a-task-3286ebb39f40

Reply:
Interesting. I didn't consider ignore_errors because I'm not ignoring an actual error that ansible would usually detect, just noticing error messages in a log file. I had not considered using a `fail` task with ignore_errors; what a silly and useful idea. I'll try it! The output is a bit confusing IMO, but probably still better.

Reply:
Regarding the callback module, I made a custom one for stf functional tests, which produces a report at the end that summarises the tests

Reply:
I've already worked out using the rescue block and flag setting to let tests fail but still continue with the playbook, that's pretty sorted. This "Check for errors" block is not really about determining success or failure of tests, but rather to bring attention to errors in the log file that may be present despite your tests passing. I'll take a look at that custom logger callback. Thanks!